model/task_lifecycle.go

```diff
@@ -518,14 +518,29 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		return errors.Wrap(err, "marking task finished")
 	}
 
-	if err = UpdateUnblockedDependencies(t, true, "MarkEnd"); err != nil {
-		return errors.Wrap(err, "updating unblocked dependencies")
+	unblockedIds := []string{}
+	if t.IsPartOfSingleHostTaskGroup() {
+		unblockedIds, err = UpdateUnblockedDependencies(t)
+		if err != nil {
+			return errors.Wrap(err, "updating unblocked dependencies")
+		}
 	}
 
-	if err = UpdateBlockedDependencies(t); err != nil {
+	blockedIds, err := UpdateBlockedDependencies(t)
+	if err != nil {
 		return errors.Wrap(err, "updating blocked dependencies")
 	}
 
+	unblockedAndNotBlockedIds, _ := utility.StringSliceSymmetricDifference(unblockedIds, blockedIds)
+	// The question to determine here is: do we ever unblock more dependencies than we block?
```

Comment:
I don't quite understand this comment, can you clarify it a bit? The log seems to trigger when we unblock dependencies that remain unblocked after updating blocked dependencies, which is not dependent on the relative number of dependencies we block/unblock.

Should this be phrased as something like "do we ever unblock dependencies that don't immediately get re-blocked"?

Maybe I don't understand this log, since it seems correct to me that a previously-unattainable dependency might be unblocked because it's now attainable.

Reply:
I'm fine to re-phrase the comment.